### PR TITLE
Implemented new scripts and tests

### LINKED ISSUE #9: Implementation of extra functionality from lodash
The following functions are in lodash that seem useful in GM context:

_chunk
_difference
_drop
_fill
_intersection
_pull
_reverse
_slice
_union
_without
_zip

### PR DESCRIPTION
**Pull request by trouvant**
Resolves #9.

Made the following changes to existing scripts:
- Renamed `_reverse` (i.e. the non-mutative function) to `_backward`, modelled after the tendency in Lodash for non-mutative versions of functions to be named adjectivally or nominally, whereas mutative counterparts are typically named with verbs. (Let me know if you approve of this solution, or if you would rather use `_reverse` and `reverse_mut`.)
- Optimized the array-creation loop in `_backward` (formerly `_reverse`).
- Replaced the test for `_reverse` in Test Suite 2 with a similar test for `_backward`.

Implemented the following scripts:
- `_chunk`
- `_difference`
- `_difference_by`
- `_drop`
- `_drop_right`
- `_fill`
- `_intersection`
- `_intersection_by`
- `_reverse` (which is mutative, as in Lodash)
- `_union`
- `_union_by`
- `_unzip`
- `_without`
- `_zip`

Tests for each of the above scripts have been implemented in Test Suite 4.

As noted previously, unless I am mistaken, `_pull` cannot be implemented in GML with functionality resembling that of the same function in Lodash, for GML does not natively support resizing an array except to expand it, and with no way to access the memory on the heap associated with an array, it seems impossible to me. I recommend simply using `_without` or a `ds_list` for similar behaviour.

Since my last pull request, I have corrected inconsistencies in documentation, optimized array creation, and made my variable names adherent to the contribution standards. I have also replaced usage of `div` and `mod` with the symbols `/` and `%`, with the exception of `div` used in `_reverse`: in this case, it is shorthand for integer division, and eliminates the need for `floor`. Lastly, I don't have nodejs installed, so the README will need to be updated by a maintainer.

### CODE REVIEW

**Comment by twisterghost** [comment]
Looking really good. I'll pull this down today and review it, hope to get it merged by the end of the day.

I'm debating with myself if this is a minor or major update. If the current implementation of `_reverse` is now mutative, but still returns the value passed in, it seems like a breaking API change, though the input/output is still the same so it could also be considered minor. Will think more about it.

**Comment by twisterghost** [comment]
Sorry, still tracking this. It's been a hectic few days. Will pull it down right now and plan to merge it (should everything seem ok) tomorrow morning.

I'll also be marking this as release 5.0.0 since the `_reverse` is a breaking change.

File: src/scripts/_fill/_fill.gml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// => [0, 4, 4, 3];
+
+var arr = [0, 1, 2, 3];
+_fill(arr, 0);
+// => [0, 0, 0, 0];
+*/
+
+var i;
+var arr;
+var _end;
```
Comment (twisterghost) [inline src/scripts/_fill/_fill.gml]: Would be good to rename this. It's not unlikely that in the future there would be a script called `_end`.

**Comment by twisterghost** [comment]
Left one code review comment, and would also suggest changing the utility function you added for the test (`_floor`) to be something without a preceeding underscore to differentiate between gdash functions and everything else. Could call it `native_floor` or something.

Let me know if you don't have the time to make those changes. I can make them myself since they're so minor. Otherwise, I'll merge this when those changes are made.

Thank you!

**Comment by trouvant** [comment]
No problem! I've taken care of those changes and ran the tests again, just to be sure. Let me know if there's anything else I should update.

**Comment by twisterghost** [comment]
Perfect! Will merge shortly.

**Pull request merged**